### PR TITLE
allow null TargetHost in SslClientAuthenticationOptions

### PR DESCRIPTION
**Pull request by wfurt**
This is follow-up on  #68678.
- There seems to be no good reason to throw if TargetHost is not set. We already support empty string and SNI is optional.
- _sslAuthenticationOptions are no longer nullable so there is no need for special annotation.

### CODE REVIEW

File: src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs [hunk 1]
```diff
@@ -49,7 +49,10 @@ internal void UpdateOptions(SslClientAuthenticationOptions sslClientAuthenticati
             IsServer = false;
             RemoteCertRequired = true;
             // RFC 6066 section 3 says to exclude trailing dot from fully qualified DNS hostname
-            TargetHost = sslClientAuthenticationOptions.TargetHost.TrimEnd('.');
+            if (sslClientAuthenticationOptions.TargetHost != null)
```
Comment (stephentoub) [inline src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs:52]: There's an assert on L20 that this is not null.

**Comment by stephentoub** [comment]
Tests?

File: src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs (hunk 1 above)
Comment (wfurt) [reply src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs:50]: updated and I added new test for this case.

**Review by rzikm — APPROVED** [review]
LGTM

**Pull request merged**